daemon/api_snaps_test.go

```diff
@@ -1135,25 +1168,17 @@ func (s *snapsSuite) testPostSnap(c *check.C, withChannel bool) {
 
 	checked := false
 	defer daemon.MockSnapstateInstall(func(ctx context.Context, s *state.State, name string, opts *snapstate.RevisionOptions, userID int, flags snapstate.Flags) (*state.TaskSet, error) {
-		if withChannel {
-			// channel in -> channel out
-			c.Check(opts.Channel, check.Equals, "xyzzy")
-		} else {
-			// no channel in -> no channel out
-			c.Check(opts.Channel, check.Equals, "")
-		}
+		checkOpts(opts)
```

Comment:
(nitpick^2) maybe worth checking for `if checkOpts != nil { checkOpts(opts) }` here to avoid the need to sometimes pass an empty checkOpts function but I guess it's a bit of a wash if the "complicity" is in here or if it is on the caller side.

Reply:
I can change this in a follow-up